crypto/hmac_extra/hmac_test.cc

```diff
@@ -227,20 +235,71 @@ TEST(HMACTest, TestVectors) {
+    // and the precomputed_key_export flag cannot be set
+    bssl::ScopedHMAC_CTX ctx2;
+    ASSERT_FALSE(HMAC_set_precomputed_key_export(ctx2.get()));
+    precomputed_key_len_out = HMAC_MAX_PRECOMPUTED_KEY_SIZE;
+    ASSERT_FALSE(HMAC_get_precomputed_key(ctx2.get(), precomputed_key, &precomputed_key_len_out));
+
+    // Get the precomputed key length for later use
+    // And test the precomputed key size is at most HMAC_MAX_PRECOMPUTED_KEY_SIZE
+    ASSERT_TRUE(HMAC_set_precomputed_key_export(ctx.get()));
+    size_t precomputed_key_len;
+    HMAC_get_precomputed_key(ctx.get(), nullptr, &precomputed_key_len);
+    ASSERT_LE(precomputed_key_len, (size_t) HMAC_MAX_PRECOMPUTED_KEY_SIZE);
+
+    // Test that at this point, the context can be used with HMAC_Update
+    ASSERT_FALSE(HMAC_Update(ctx.get(), input.data(), input.size()));
+    ASSERT_FALSE(HMAC_Final(ctx.get(), mac.get(), &mac_len));
+
+    // Export the precomputed key for later use
+    precomputed_key_len_out = HMAC_MAX_PRECOMPUTED_KEY_SIZE;
+    ASSERT_TRUE(HMAC_get_precomputed_key(ctx.get(), precomputed_key, &precomputed_key_len_out));
+    ASSERT_EQ(precomputed_key_len, precomputed_key_len_out);
+
+    // Test that at this point, the context can be used with HMAC_Update
+    ASSERT_TRUE(HMAC_Update(ctx.get(), input.data(), input.size()));
+    ASSERT_TRUE(HMAC_Final(ctx.get(), mac.get(), &mac_len));
+    EXPECT_EQ(Bytes(output), Bytes(mac.get(), mac_len));
+    OPENSSL_memset(mac.get(), 0, expected_mac_len); // Clear the prior correct answer
+
     // Test that an HMAC_CTX may be reset with the same key.
     ASSERT_TRUE(HMAC_Init_ex(ctx.get(), nullptr, 0, digest, nullptr));
     ASSERT_TRUE(HMAC_Update(ctx.get(), input.data(), input.size()));
     ASSERT_TRUE(HMAC_Final(ctx.get(), mac.get(), &mac_len));
     EXPECT_EQ(Bytes(output), Bytes(mac.get(), mac_len));
     OPENSSL_memset(mac.get(), 0, expected_mac_len); // Clear the prior correct answer
 
-    // Test that an HMAC_CTX may be reset with the same key and an null md
+    // Same test but with HMAC_Init_from_precomputed_key
+    ASSERT_TRUE(HMAC_Init_from_precomputed_key(ctx.get(), nullptr, 0, digest));
+    ASSERT_TRUE(HMAC_Update(ctx.get(), input.data(), input.size()));
+    ASSERT_TRUE(HMAC_Final(ctx.get(), mac.get(), &mac_len));
+    EXPECT_EQ(Bytes(output), Bytes(mac.get(), mac_len));
+    OPENSSL_memset(mac.get(), 0, expected_mac_len); // Clear the prior correct answer
+
+    // Test that an HMAC_CTX may be reset with the same key and a null md
     ASSERT_TRUE(HMAC_Init_ex(ctx.get(), nullptr, 0, nullptr, nullptr));
     ASSERT_TRUE(HMAC_Update(ctx.get(), input.data(), input.size()));
     ASSERT_TRUE(HMAC_Final(ctx.get(), mac.get(), &mac_len));
     EXPECT_EQ(Bytes(output), Bytes(mac.get(), mac_len));
     OPENSSL_memset(mac.get(), 0, expected_mac_len);  // Clear the prior correct answer
 
+    // Same test but using the Init_with_precompute instead
```

Comment:
```suggestion
    // Same test but using the Init_from_precomputed_key instead
```
Would you like to change to this in all comments?